### PR TITLE
Sanitise dependencies

### LINKED ISSUE #3795: Remove unrequited dependencies
A sample application when it pulls in the `composer-client` module brings with it 666 dependencies and about 50Mb of code.

A substantial amount is from duplicated `composer-common`.  

We can take advantage of the peerDependancy concept in npm to try and remove some of this.

### PR DESCRIPTION
**Pull request by mbwhite**
Introduce peerDependencies for a number of core modules to reduce the size of them. 

part resolves #3795

### CODE REVIEW

File: packages/composer-runtime-embedded/package.json [hunk 1]
```diff
@@ -72,8 +72,10 @@
       }
     }
   },
+  "peerDependencies": {
+    "composer-common": "0.19.1"
+  },  
   "dependencies": {
-    "composer-common": "0.19.1",
```
Comment (sstone1) [inline packages/composer-runtime-embedded/package.json]: Why is not in `devDependencies` like the other ones?

File: packages/composer-wallet-filesystem/package.json [hunk 2]
```diff
@@ -76,8 +76,10 @@
       }
     }
   },
+  "peerDependencies": {
+    "composer-common": "0.19.1"
+  },
   "dependencies": {
-    "composer-common": "0.19.1",
```
Comment (sstone1) [inline packages/composer-wallet-filesystem/package.json]: Why is not in devDependencies like the other ones?

File: packages/composer-wallet-inmemory/package.json [hunk 3]
```diff
@@ -76,10 +76,10 @@
       }
     }
   },
+  "peerDependencies": {
+    "composer-common": "0.19.1"
+  },
   "dependencies": {
-    "composer-common": "0.19.1",
```
Comment (sstone1) [inline packages/composer-wallet-inmemory/package.json]: Why is not in devDependencies like the other ones?

**Comment by mbwhite** [comment]
@sstone1 
good question; devDepenendencies need to have composer-common when whatever function the tests invoke, actually need `composer-common`. In the cases you've mentioned - these tests are not invoking composer-common code so don't need the devDedependency.

**Comment by sstone1** [comment]
This doesn't build @mbwhite!

**Comment by mbwhite** [comment]
@sstone1 not yet no...

**Comment by mbwhite** [comment]
Build works on my own travis org... https://travis-ci.org/mbwhite/composer/builds/371939672

**Pull request merged**